src/esssans/normalization.py

```diff
@@ -64,6 +64,41 @@ def solid_angle_rectangular_approximation(
     )
 
 
+def solid_angle_approximation_for_cylinders(cylinders: sc.DataSet, origin: sc.Variable):
```

Comment:
`origin` here might be a bad name, it's the "point of view" from where the solid angle is observed. In our case I guess we would pass the sample position there.

For this function to work we need that `face_center` and `face_edge` and the origin are points in the same (orthonormal) coordinate system. To make that the case I think we will need the cylinder spec + cylinder positions + sample position like you say. But my thinking was that this could be something handled by the caller.